### ISSUE TITLE
vmware clone task fails instead of waiting to completion

### ISSUE THREAD

**Issue by nghgd** [saltstack/salt#28511]
The task fails immediately, even though the clone process is still in progress.

This occurs randomly, sometimes it works fine and sometimes it fails and I've not been able to find a reason for it.

```
sudo salt-cloud -l debug -p <profile> <machine>
```

Result when it errors:
```
[ERROR   ] Error creating <machine>: None
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/vmware.py", line 2446, in create
    _wait_for_task(task, vm_name, "clone", 5, 'info')
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/vmware.py", line 823, in _wait_for_task
    raise Exception(task.info.error)
Exception: None
[DEBUG   ] LazyLoaded nested.output
<machine>:
    ----------
    Error:
        Error creating <machine>: None
```

```
salt --versions-report
Salt Version:
           Salt: 2015.8.1

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: 0.9.1
         PyYAML: 3.10
          PyZMQ: 14.0.1
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.4
           cffi: Not Installed
       cherrypy: 3.2.2
       dateutil: 1.5
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-65-generic
         system: Ubuntu 14.04 trusty
```

**Comment by nghgd** [comment]
I'd also like to note that this seems to affect other tasks...including destroy.

```
[ERROR   ] Error while destroying VM <machine>: None
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/vmware.py", line 2133, in destroy
    _wait_for_task(task, name, "destroy")
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/vmware.py", line 823, in _wait_for_task
    raise Exception(task.info.error)
Exception: None
[DEBUG   ] Deleted '/etc/salt/pki/master/minions/<machine>'
[DEBUG   ] LazyLoaded nested.output
vmware:
    ----------
    vmware:
        ----------
        <machine>:
            failed to destroy
```

**Comment by jfindlay** [comment]
@nghgd, thanks for the report.  Ping @nmadhok.

**Comment by nmadhok** [comment]
@jfindlay I will look into this. Haven't come across this yet myself. Can you assign me to this issue and other VMware cloud driver related issues in future? It would make it easier for me to keep track of assigned issues.

**Comment by nmadhok** [comment]
@nghgd Does the vCenter show the task as running or completed after you get this failure?

**Comment by nghgd** [comment]
Running

**Comment by nmadhok** [comment]
@nghgd The referenced commit above will fix the issue you're having.

**Comment by jfindlay** [comment]
@nmadhok, github is not allowing me to assign the issue to you, but it seems that you were able to fix it anyway, thanks.

**Comment by DanyC97** [comment]
@nmadhok  i'm afraid even on the latest 2015.8.8.2 i'm still getting the same issue

How do you debug it? apart from the `-l debug` ?

**Comment by jpl154** [comment]
@nmadhok I too am still experiencing this issue using the vmware driver for vSphere in 2016.3.3

**Comment by danielpalstra** [comment]
@nmadhok same issue here with version 2015.3.4 and `pyvmomi==6.0.0.2016.4`

**Comment by jbfriedrich** [comment]
I still have this problem in 2016.3.4 with VMware vSphere 5.5. The task.info.state is still "running", I checked that, but it seems that the "while loop" is prematurely evacuated. I cannot see the reason for it though. I tried different sleep times, diffferent sleep seconds, but to no avail.

Should I open a new issue as this one was fixed with PR #28546?
